sql/descriptor_cache.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Marc Berhault (marc@cockroachlabs.com)
+
+package sql
+
+import (
+	"fmt"
+
+	"github.com/cockroachdb/cockroach/sql/parser"
+	gogoproto "github.com/gogo/protobuf/proto"
+)
+
+// TestingDisableDescriptorCache is used by tests and benchmarks to disable
+// the descriptor cache functionality. All lookups will go straight through
+// the DB client.
+var TestingDisableDescriptorCache bool
+
+// getCachedTableDesc looks for a table descriptor in the descriptor cache.
+func (p *planner) getCachedTableDesc(qname *parser.QualifiedName) (*TableDescriptor, error) {
+	if err := qname.NormalizeTableName(p.session.Database); err != nil {
+		return nil, err
+	}
+	dbDesc, cached, err := p.getCachedDatabaseDesc(qname.Database())
+	if err != nil {
+		return nil, err
+	}
+
+	desc := TableDescriptor{}
```

Comment:
The compiler will heap allocate this object because you're taking a pointer to it later. I'm a fan of being explicit and declaring this as a pointer in the first place:

```
  desc := &TableDescriptor{}
```